### ISSUE TITLE
Add support for Windows Authentication

### ISSUE THREAD

**Issue by wsmelton** [thycotic-ps/thycotic.secretserver#24]
**Is your feature request related to a problem? Please describe.**

Add support to `New-TssSession` for Windows Authentication with IWA.

**Comment by peetrike** [comment]
Is there anything done with this?

**Comment by wsmelton** [comment]
You can see the status of issues and work via the [Projects](https://github.com/thycotic-ps/thycotic.secretserver/projects) tab of the repository.

**Comment by peetrike** [comment]
Would you mind me making PR for it?  It should be simple to add...

**Comment by wsmelton** [comment]
Actually supporting WinAuth is not simply adding the URL, just as an FYI because not all REST API endpoints are available to the `winauthwebservices/api/v1` path.

You are welcome to create a PR though if you work it out, the contributing guide has details on the testing setup. As the command to work on this affects the module as a whole. Each test for a function will also need to be modified to include testing both endpoint access types.

**Comment by wsmelton** [comment]
I have plans to get a part script implemented for validating the version of Secret Server and it may need to be expanded to also cover outputting whether a given endpoint is not there for winauth 🤔 .

**Comment by peetrike** [comment]
> Actually supporting WinAuth is not simply adding the URL, just as an FYI because not all REST API endpoints are available to the `winauthwebservices/api/v1` path.

And every single function should be changed, because Invoke-RestApi doesn't take TssSession object as input parameter.  Even if it would, all functions still need to be changed ...

**Comment by wsmelton** [comment]
...so not a simply change

**Comment by wsmelton** [comment]
TssSession will always be a required parameter on the functions. That won't change.

**Comment by peetrike** [comment]
Yes.  I'm still interested of the feature, so let me at least think how it could be integrated :)

**Comment by peetrike** [comment]
> TssSession will always be a required parameter on the functions. That won't change.

I'm more about thinking how to add -TssSession to Invoke-RestApi.  so that it doesn't break existing code...

**Comment by wsmelton** [comment]
There is no reason to touch `Invoke-TssRestApi`. It does not need any changes to support WinAuth.

**Comment by peetrike** [comment]
i know.  But otherwise all functions would have to deal with new type of session.  Or a new "part" should be added, that would deal creating 'Invoke-TssRestApi' call.

**Comment by wsmelton** [comment]
Sorry but `Invoke-TssRestApi` is off-limits. 

Parts are basically treated as private functions.

**Comment by wsmelton** [comment]
All of this work, off the top of my head (in theory) should be able to deal with this from `New-TssSession` processing the session object.

**Comment by peetrike** [comment]
in the beginning i thought that I need to change `New-TssSession` to emit session with new TokenType, and change TssSession class.  But every other function calls `Invoke-TssRestApi` and they shoud be able to add correct parameters to that call.

**Comment by wsmelton** [comment]
Adjusting the token type would be one method to use that would allow us in the functions to validate that when additional functions are added that do not have an equivalent `winauthwebservices` endpoint.

The method in which the parameters are being passed in each function to `Invoke-TssRestApi` should be irrelevant because it will handle the token missing if it was not found. Pester tests should catch this currently though I believe if it would be an issue or not.

**Comment by peetrike** [comment]
meanwhile, found small difference in TssSession class and New-TssSession function: https://github.com/thycotic-ps/thycotic.secretserver/pull/73.  

Should I also open issue to mark that?

**Comment by peetrike** [comment]
I'll create a PR and put only changes to `New-TssSession` and TssSession class (plus tests).  Then the changes to functions using that type of token could be filed in later.

**Comment by wsmelton** [comment]
Checked the OpenAPI docs for 10.9.33 and I'm not catching any differences between the endpoints, so there may be no need to touch the functions to validate anything.

**Comment by wsmelton** [comment]
Worked over the weekend and finally got this working; had to change a good bit on how the tests and commands were run. I did find a possible bug in the endpoints used by `Set-TssSecret`. Those are being skipped when Windows Auth is used running tests for now until I can work out why 404 error is being thrown (could be my local lab setup, but I will test in our main lab this week sometime).

![image](https://user-images.githubusercontent.com/11204251/107965703-72ebac00-6f70-11eb-819b-8bc90025141c.png)

I've got one test that continues to spit the warning out, will circle back around to fix that at a later time.
